### ISSUE TITLE
ifort fails  to compile source .f files with .h files

### ISSUE THREAD

**Issue by jirikanka** [opencollab/arpack-ng#139]
I need to build a static library libarpack.a for the DFTB+ semi-empirical DFT code. When I changed to the SRC directory and issued the command
ifort -c -O2 *.f    (ifort 2018.3.222)
it ended up with (syntax) error messages coming from the included stat.h and debug.h files.
For illustration I attach below the error messages generated by a failed compilation of cgetv0.f file.
Could you please help me to resolve the compilation problem (using some additional ifort option or modifying stat.h and debug.h files?).
```

debug.h(8): error #5082: Syntax error, found '&' when expecting one of: <IDENTIFIER> %FILL
      integer  logfil, ndigit, mgetv0,                                &
----------------------------------------------------------------------^
debug.h(9): error #5082: Syntax error, found ',' when expecting one of: => = . [ % ( :
               msaupd, msaup2, msaitr, mseigt, msapps, msgets, mseupd,&
---------------------^
debug.h(9): error #5082: Syntax error, found '&' when expecting one of: <REAL_KIND_CON> <REAL_CONSTANT> <DBLPRC_CONSTANT> <DBLPRC_KIND_CON> <QUAPRC_CONSTANT> ...
               msaupd, msaup2, msaitr, mseigt, msapps, msgets, mseupd,&
----------------------------------------------------------------------^
debug.h(10): error #5082: Syntax error, found ',' when expecting one of: => = . [ % ( :
               mnaupd, mnaup2, mnaitr, mneigh, mnapps, mngets, mneupd,&
---------------------^
debug.h(10): error #5082: Syntax error, found '&' when expecting one of: <REAL_KIND_CON> <REAL_CONSTANT> <DBLPRC_CONSTANT> <DBLPRC_KIND_CON> <QUAPRC_CONSTANT> ...
               mnaupd, mnaup2, mnaitr, mneigh, mnapps, mngets, mneupd,&
----------------------------------------------------------------------^
debug.h(11): error #5082: Syntax error, found ',' when expecting one of: => = . [ % ( :
               mcaupd, mcaup2, mcaitr, mceigh, mcapps, mcgets, mceupd
---------------------^
debug.h(12): error #5082: Syntax error, found '&' when expecting one of: <IDENTIFIER>
      common /debug/                                                  &
----------------------------------------------------------------------^
debug.h(13): error #5082: Syntax error, found ',' when expecting one of: => = . [ % ( :
               logfil, ndigit, mgetv0,                                &
---------------------^
debug.h(13): error #5082: Syntax error, found '&' when expecting one of: <REAL_KIND_CON> <REAL_CONSTANT> <DBLPRC_CONSTANT> <DBLPRC_KIND_CON> <QUAPRC_CONSTANT> ...
               logfil, ndigit, mgetv0,                                &
----------------------------------------------------------------------^
debug.h(14): error #5082: Syntax error, found ',' when expecting one of: => = . [ % ( :
               msaupd, msaup2, msaitr, mseigt, msapps, msgets, mseupd,&
---------------------^
debug.h(14): error #5082: Syntax error, found '&' when expecting one of: <REAL_KIND_CON> <REAL_CONSTANT> <DBLPRC_CONSTANT> <DBLPRC_KIND_CON> <QUAPRC_CONSTANT> ...
               msaupd, msaup2, msaitr, mseigt, msapps, msgets, mseupd,&
----------------------------------------------------------------------^
debug.h(15): error #5082: Syntax error, found ',' when expecting one of: => = . [ % ( :
               mnaupd, mnaup2, mnaitr, mneigh, mnapps, mngets, mneupd,&
---------------------^
debug.h(15): error #5082: Syntax error, found '&' when expecting one of: <REAL_KIND_CON> <REAL_CONSTANT> <DBLPRC_CONSTANT> <DBLPRC_KIND_CON> <QUAPRC_CONSTANT> ...
               mnaupd, mnaup2, mnaitr, mneigh, mnapps, mngets, mneupd,&
----------------------------------------------------------------------^
debug.h(16): error #5082: Syntax error, found ',' when expecting one of: => = . [ % ( :
               mcaupd, mcaup2, mcaitr, mceigh, mcapps, mcgets, mceupd
---------------------^
stat.h(12): error #5082: Syntax error, found END-OF-STATEMENT when expecting one of: <IDENTIFIER> %FILL
      real       tsaupd, tsaup2, tsaitr, tseigt, tsgets, tsapps, tsconv,&
-------------------------------------------------------------------------^
stat.h(13): error #5082: Syntax error, found END-OF-STATEMENT when expecting one of: <IDENTIFIER> %FILL
                 tnaupd, tnaup2, tnaitr, tneigh, tngets, tnapps, tnconv,&
-------------------------------------------------------------------------^
stat.h(14): error #5082: Syntax error, found END-OF-STATEMENT when expecting one of: <IDENTIFIER> %FILL
                 tcaupd, tcaup2, tcaitr, tceigh, tcgets, tcapps, tcconv,&
-------------------------------------------------------------------------^
stat.h(16): error #5082: Syntax error, found END-OF-STATEMENT when expecting one of: <IDENTIFIER>
      common /timing/                                                   &
-------------------------------------------------------------------------^
stat.h(17): error #5082: Syntax error, found END-OF-STATEMENT when expecting one of: // / <IDENTIFIER>
                 nopx, nbx, nrorth, nitref, nrstrt,                     &
-------------------------------------------------------------------------^
stat.h(18): error #5082: Syntax error, found END-OF-STATEMENT when expecting one of: // / <IDENTIFIER>
                 tsaupd, tsaup2, tsaitr, tseigt, tsgets, tsapps, tsconv,&
-------------------------------------------------------------------------^
stat.h(19): error #5082: Syntax error, found END-OF-STATEMENT when expecting one of: // / <IDENTIFIER>
                 tnaupd, tnaup2, tnaitr, tneigh, tngets, tnapps, tnconv,&
-------------------------------------------------------------------------^
stat.h(20): error #5082: Syntax error, found END-OF-STATEMENT when expecting one of: // / <IDENTIFIER>
                 tcaupd, tcaup2, tcaitr, tceigh, tcgets, tcapps, tcconv,&
-------------------------------------------------------------------------^
debug.h(16): error #6418: This name has already been assigned a data type.   [MCAUPD]
               mcaupd, mcaup2, mcaitr, mceigh, mcapps, mcgets, mceupd
---------------^
debug.h(16): error #6418: This name has already been assigned a data type.   [MCAUP2]
               mcaupd, mcaup2, mcaitr, mceigh, mcapps, mcgets, mceupd
-----------------------^
debug.h(16): error #6418: This name has already been assigned a data type.   [MCAITR]
               mcaupd, mcaup2, mcaitr, mceigh, mcapps, mcgets, mceupd
-------------------------------^
debug.h(16): error #6418: This name has already been assigned a data type.   [MCEIGH]
               mcaupd, mcaup2, mcaitr, mceigh, mcapps, mcgets, mceupd
---------------------------------------^
debug.h(16): error #6418: This name has already been assigned a data type.   [MCAPPS]
               mcaupd, mcaup2, mcaitr, mceigh, mcapps, mcgets, mceupd
-----------------------------------------------^
debug.h(16): error #6418: This name has already been assigned a data type.   [MCGETS]
               mcaupd, mcaup2, mcaitr, mceigh, mcapps, mcgets, mceupd
-------------------------------------------------------^
debug.h(16): error #6418: This name has already been assigned a data type.   [MCEUPD]
               mcaupd, mcaup2, mcaitr, mceigh, mcapps, mcgets, mceupd
---------------------------------------------------------------^
compilation aborted for cgetv0.f (code 1)
```

**Comment by jirikanka** [comment]
The compilation problem has been fixed by changing the line continuation syntax from F90 to F77 in stat.h and debug.h files.

**Comment by sylvestre** [comment]
Don't hesitate to submit a pull request!

**Comment by caliarim** [comment]
It seems to me that now debug.h and stat.h have F77 continuation syntax. Can we close this issue?

**Comment by fghoussen** [comment]
Closing as likely dup/fixed of/by #339. Do not hesitate to reopen if the problem is still here.
